integration/tests/bft/topicvalidator/orchestrator.go

```diff
@@ -0,0 +1,275 @@
+			TargetNum:           0,
+			TargetIds:           flow.IdentifierList{o.victimEN},
+			FlowProtocolEvent:   unauthorizedProposal,
+			FlowProtocolEventID: eventID,
+		}
+		o.unauthorizedEvents[eventID] = unauthorizedMsgByRole
+	}
+}
+
+// initUnauthorizedMsgOnChannelEvents sets n number of events where the message is not
+// authorized to be sent on the event channel.
+func (o *testOrchestrator) initUnauthorizedMsgOnChannelEvents(n int) {
+	for i := 0; i < n; i++ {
+		syncReq := &messages.SyncRequest{
+			Nonce:  rand.Uint64(),
+			Height: rand.Uint64(),
+		}
+		eventID := o.getFlowProtocolEventID(syncReq)
+		unauthorizedMsgOnChannel := &insecure.EgressEvent{
+			CorruptOriginId:     o.attackerAN,
+			Channel:             channels.PushReceipts,
+			Protocol:            insecure.Protocol_PUBLISH,
+			TargetNum:           0,
+			TargetIds:           flow.IdentifierList{o.victimEN},
+			FlowProtocolEvent:   syncReq,
+			FlowProtocolEventID: eventID,
+		}
+		o.unauthorizedEvents[eventID] = unauthorizedMsgOnChannel
+	}
+}
+
+// initUnauthorizedUnicastOnChannelEvents sets n number of events where the message is not
+// authorized to be sent via insecure.Protocol_UNICAST on the event channel.
+func (o *testOrchestrator) initUnauthorizedUnicastOnChannelEvents(n int) {
+	for i := 0; i < n; i++ {
+		syncReq := &messages.SyncRequest{
+			Nonce:  rand.Uint64(),
+			Height: rand.Uint64(),
+		}
+		eventID := o.getFlowProtocolEventID(syncReq)
+		unauthorizedUnicastOnChannel := &insecure.EgressEvent{
+			CorruptOriginId:     o.attackerAN,
+			Channel:             channels.SyncCommittee,
+			Protocol:            insecure.Protocol_UNICAST,
+			TargetNum:           0,
+			TargetIds:           flow.IdentifierList{o.victimEN},
+			FlowProtocolEvent:   syncReq,
+			FlowProtocolEventID: eventID,
+		}
+		o.unauthorizedEvents[eventID] = unauthorizedUnicastOnChannel
+	}
+}
+
+// initUnauthorizedPublishOnChannelEvents sets n number of events where the message is not
+// authorized to be sent via insecure.Protocol_PUBLISH on the event channel.
+func (o *testOrchestrator) initUnauthorizedPublishOnChannelEvents(n int) {
+	for i := 0; i < n; i++ {
+		chunkDataResponse := unittest.ChunkDataResponseMsgFixture(unittest.IdentifierFixture())
+		eventID := o.getFlowProtocolEventID(chunkDataResponse)
+		unauthorizedPublishOnChannel := &insecure.EgressEvent{
+			CorruptOriginId:     o.attackerEN,
+			Channel:             channels.ProvideChunks,
+			Protocol:            insecure.Protocol_PUBLISH,
+			TargetNum:           0,
+			TargetIds:           flow.IdentifierList{o.victimVN},
+			FlowProtocolEvent:   chunkDataResponse,
+			FlowProtocolEventID: eventID,
+		}
+		o.unauthorizedEvents[eventID] = unauthorizedPublishOnChannel
+	}
+}
+
+func (o *testOrchestrator) getFlowProtocolEventID(event interface{}) flow.Identifier {
+	payload, err := o.codec.Encode(event)
+	require.NoError(o.t, err)
+	return flow.HashToID(hash.DefaultHasher.ComputeHash(payload))
```

Comment:
```suggestion
	return flow.HashToID(hash.DefaultHasher.ComputeHash(payload))
```
Please also use `EventId` [method](https://github.com/onflow/flow-go/blob/master/network/p2p/network.go#L528-L528).

Reply:
https://github.com/onflow/flow-go/pull/3580/commits/3c84a08a7331610b83cc9981f90f9c24acd326ed

Reply:
Please keep `FlowProtocolEventID` as a `flow.Identifier`. Instead use the [`HexStringToIdentifier`](https://github.com/onflow/flow-go/blob/50a959a101a65ea37bf520b1adf60db934089949/model/flow/identifier.go#L39) here to convert the `hash` object into a `flow.Identifier`.